stixcore/processing/publish.py

```diff
@@ -0,0 +1,300 @@
+        try:
+            self.cur.execute("insert into published values (?, ?, ?, ?, ?, ?)",
+                             (name, dir, version, date, m_date, esa_name))
+            others = self.find_by_esa_name(esa_name)
+            if (len(others) > 1):
+                return (PublishConflicts.SAME_ESA_NAME, others)
+            else:
+                return (PublishConflicts.ADDED, others)
+        except sqlite3.IntegrityError:
+            return (PublishConflicts.SAME_EXISTS, self.find_by_name(name))
+
+    def _execute(self, sql, arguments=None, result_type='list'):
+        """Execute sql and return results in a list or a dictionary."""
+        if not self.cur:
+            raise Exception('DB is not initialized!')
+        else:
+            if arguments:
+                self.cur.execute(sql, arguments)
+            else:
+                self.cur.execute(sql)
+            if result_type == 'list':
+                rows = self.cur.fetchall()
+            else:
+                rows = [
+                    dict(
+                        zip([column[0]
+                            for column in self.cur.description], row))
+                    for row in self.cur.fetchall()
+                ]
+            return rows
+
+    def find_by_name(self, name):
+        return self._execute("select * from published where name = ?", (name, ), result_type='hash')
+
+    def find_by_esa_name(self, esa_name):
+        return self._execute("select * from published where esaname = ?", (esa_name, ),
+                             result_type='hash')
+
+    def get_all(self):
+        return self._execute("select * from published")
+
+
+def send_mail_report(files):
+    if CONFIG.getboolean('Publish', 'report_mail_send', fallback=False):
+        try:
+            sender = CONFIG.get('Pipeline', 'error_mail_sender', fallback='')
+            receivers = CONFIG.get('Publish', 'report_mail_receivers').split(",")
+            host = CONFIG.get('Pipeline', 'error_mail_smpt_host', fallback='localhost')
+            port = CONFIG.getint('Pipeline', 'error_mail_smpt_port', fallback=25)
+            smtp_server = smtplib.SMTP(host=host, port=port)
+            summary = (f"E {len(files['failed'])} "
+                       f"C {len(files[PublishConflicts.SAME_EXISTS])} "
+                       f"D {len(files[PublishConflicts.SAME_ESA_NAME])} "
+                       f"A {len(files[PublishConflicts.ADDED])}")
+            error = "" if 'failed' not in files else "ERROR-"
+            message = f"""Subject: StixCore TMTC Publishing {error}Report {summary}
+
+ERRORS
+******
+{files['failed']}
+
+PUBLISHED DUPLICATES
+********************
+{pformat(files[PublishConflicts.SAME_EXISTS])}
+
+PUBLISHED ESA DUPLICATES
+************************
+{pformat(files[PublishConflicts.SAME_ESA_NAME])}
+
+PUBLISHED
+*********
+{pformat(files[PublishConflicts.ADDED])}
+
+
+StixCore
+
+==========================
+
+do not answer to this mail.
+"""
```

Comment:
move to var defined at top of file may `PUBLISH_EMAIL_TEMPLATE = f""" ...."""`

Reply:
good point